### PR TITLE
AnimePahe Tweaks

### PR DESCRIPTION
**Pull request by MulverineX**
- Localization
- Anime list now doesn't break when AniList isn't present; falls back to MAL
- Anime list button now shows on playback
- Genre, Season, Medium, & Misc. Category/Page browsing support
- Moved the pause position to smallText & moved the anime title to state
- Small things

### CODE REVIEW

File: websites/A/animepahe/presence.ts [hunk 1]
```diff
@@ -77,140 +95,139 @@ function getTimes(time: number): Record<string, number> {
...
-      !movie ? `${(await strings).episode.slice(0, -3)} ${episode} of ` : ""
-    }${title}`;
+    if (!movie) presenceData.details = `Watching ${uncapitalize((await strings).episode.slice(0, -3))}${episode}`;
+    else presenceData.details = 'Watching Movie';
+
+    presenceData.state = title;
 
     if (!iframe_response.paused) {
-      presenceData.state = `${(await strings).play}`;
       presenceData.startTimestamp = timestamps[0];
```
Comment (ririxidev) [inline websites/A/animepahe/presence.ts:207]: don't need startTimestamp

File: websites/A/animepahe/presence.ts [hunk 2]
```diff
@@ -17,48 +17,66 @@ let iframe_response = {
...
+ */
+function getTimestamps(
+  videoTime: number,
+  videoDuration: number
+): Array<number> {
+  const startTime = Date.now(),
+    endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+  return [Math.floor(startTime / 1000), endTime];
+}
+
```
Comment (ririxidev) [inline websites/A/animepahe/presence.ts:79]: Use presence.getTimestamps() instead

File: websites/A/animepahe/presence.ts [hunk 3]
```diff
@@ -17,48 +17,66 @@ let iframe_response = {
...
-      localStorage.setItem("presence_data", btoa(JSON.stringify(this.list)));
+      localStorage.setItem('presence_data', btoa(JSON.stringify(this.list)));
     }
   }
 
-  constructor() {
+  constructor () {
     let storage: store_type | string | null =
-      localStorage.getItem("presence_data");
+      localStorage.getItem('presence_data');
```
Comment (ririxidev) [inline websites/A/animepahe/presence.ts:53]: You are not allowed to use localStorage, idk who allowed you to do that.

File: websites/A/animepahe/presence.ts [hunk 4]
```diff
@@ -17,48 +17,66 @@ let iframe_response = {
...
+      if (Object.keys(this.list).length === 11) {
+delete this.list[
           Object.entries(Object.assign({}, this.list)).sort(
             (a, b) => a[1].time - b[1].time
           )[0][0]
         ];
+}
 
-      localStorage.setItem("presence_data", btoa(JSON.stringify(this.list)));
+      localStorage.setItem('presence_data', btoa(JSON.stringify(this.list)));
```
Comment (ririxidev) [inline websites/A/animepahe/presence.ts:47]: same like above

**Review by rxri — CHANGES_REQUESTED** [review]
and fix codefactor issues thanks.

**Comment by MulverineX** [comment]
Holding PR until [this](https://github.com/PreMiD/Localization/pull/6668), then going to implement Localization

**Comment by MulverineX** [comment]
Holding until dumb destructuring thing is fixed & my localization strings start loading

**Comment by MulverineX** [comment]
Aight it's done, remaining bs is indent errors that ririxi told me to ignore. Now just waiting on @Timeraa to fix the API not having my localizations.

File: websites/A/animepahe/presence.ts [hunk 5]
```diff
@@ -1,63 +1,83 @@
 // Note: Developer has been working on a new website design for ages,
-//       maybe at some point he'll finish it and this will need updating.
+//       maybe at some point he"ll finish it and this will need updating.
```
Comment (Bas950) [inline websites/A/animepahe/presence.ts:2]: whut

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Comment by MulverineX** [comment]
Everything is in working order, ready to merge.

File: websites/A/animepahe/presence.ts [hunk 6]
```diff
@@ -4,60 +4,80 @@
...
-  strings = presence.getStrings({
+  waitStrings = async (lang: string) => await presence.getStrings({
     play: "presence.playback.playing",
     pause: "presence.playback.paused",
-    episode: "presence.media.info.episode",
-    browse: "presence.activity.browsing"
-  });
-
-let iframe_response = {
+    browse: "presence.activity.browsing",
```
Comment (Bas950) [inline websites/A/animepahe/presence.ts:10]: isn't there a general one for it?

Comment (MulverineX) [reply websites/A/animepahe/presence.ts:10]: Its the same string so it really doesn't matter but okay

**Comment by MulverineX** [comment]
Proof:
0Uilu3zz1R

File: websites/A/animepahe/presence.ts (hunk 6 above)
Comment (Bas950) [reply websites/A/animepahe/presence.ts]: The one you used get removed in v3

**Review by Bas950 — DISMISSED** [review]

**Review by Bas950 — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**